# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/wasm32/MacroAssembler-wasm32.h
# Commit: 28feda14fdec
# Full Hash: 28feda14fdec962ea68c26a0570ea69042a5ad27
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Add tagging for JSString* in AnyRef. r=yury
#   
#   Before this commit we would need to box a JSString* in WasmValueBox
#   when we received one from JS and converted to an AnyRef.
#   
# ==============================================================================

diff -r 85605041672d -r 28feda14fdec js/src/jit/wasm32/MacroAssembler-wasm32.h
--- a/js/src/jit/wasm32/MacroAssembler-wasm32.h	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/jit/wasm32/MacroAssembler-wasm32.h	Fri Aug 04 14:06:50 2023 +0000
@@ -436,6 +436,15 @@
   }
 
   template <typename T>
+  void unboxWasmAnyRefGCThingForGCBarrier(const T&, Register) {
+    MOZ_CRASH();
+  }
+
+  void getWasmAnyRefGCThingChunk(Register, Register) {
+    MOZ_CRASH();
+  }
+
+  template <typename T>
   void unboxObjectOrNull(const T& src, Register dest) {
     MOZ_CRASH();
   }